develop/tutorials/articles/290-configuration/03-fully-custom-system-settings-ui.markdown

```diff
@@ -0,0 +1,399 @@
+the _System Settings_ application, not your own custom application. Set up the
+component and class declaration like this:
+
+    @Component(
+        immediate = true,
+        property = {
+            "javax.portlet.name=" + ConfigurationAdminPortletKeys.SYSTEM_SETTINGS,
+            "mvc.command.name=/site_navigation_language/update_language_template_configuration"
+        },
+        service = MVCActionCommand.class
+    )
+    public class UpdateLanguageTemplateConfigurationMVCActionCommand
+        extends BaseMVCActionCommand {
+
+Specify the System Settings portlet in the portlet name property and make sure
+the MVC command name matches whatever you set in your JSPs `<portlet:
+actionURL>` name property.
+
+The logic in this class should answer the question 
+
+_what should be done with the configuration value set in the custom
+configuration screen?_
+
+The code's answer, in plain English:
+
+_set the value entered by the user (in the System Settings portlet)
+into your application's configuration object._
+
+To that end, code the `doProcessAction` and obtain a reference to the
+`ConfigurationAdmin` service:
+
+    @Override
+    protected void doProcessAction(
+            ActionRequest actionRequest, ActionResponse actionResponse)
+        throws Exception {
+
+        String ddmTemplateKey = ParamUtil.getString(
+            actionRequest, "ddmTemplateKey");
+
+        Configuration configuration = _configurationAdmin.getConfiguration(
+            SiteNavigationLanguageWebTemplateConfiguration.class.getName(),
+            StringPool.QUESTION);
+
+        Dictionary<String, Object> properties = configuration.getProperties();
+
+        if (properties == null) {
+            properties = new HashMapDictionary<>();
+        }
+
+        properties.put("ddmTemplateKey", ddmTemplateKey);
+
+        configuration.update(properties);
+    }
+
+    @Reference
+    private ConfigurationAdmin _configurationAdmin;
+
+    }
+
+This example gets the DDM template key that the user selects from the request
+and updates the configuration's properties with the key and value. Now any code
+that 
+[retrieves the configuration's values](/develop/tutorials/-/knowledge_base/7-1/reading-configuration-values-from-a-mvc-portlet)
+doesn't have to know anything about the customized UI or the
+`ConfigurationScreen` implementation.
+
+## Writing the JSP Markup [](id=writing-the-jsp-markup)
+
+Now write the JSP:
+
+    <%@ include file="/init.jsp" %>
+
+    <%
+    LanguageTemplateConfigurationDisplayContext languageTemplateConfigurationDisplayContext = (LanguageTemplateConfigurationDisplayContext)request.getAttribute(LanguageTemplateConfigurationDisplayContext.class.getName());
+
+    String currentTemplateName = languageTemplateConfigurationDisplayContext.getCurrentTemplateName();
+    %>
+
+    <portlet:actionURL name="/site_navigation_language/update_language_template_configuration" var="editURL" />
+
+    <div class="sheet sheet-lb">
```

Comment:
`sheet-lb` should probably be `sheet-lg` according to http://pat270.github.io/lexicon/content/sheet/#sheet-1